core/iwasm/common/wasm_c_api.c

```diff
@@ -467,6 +471,10 @@ wasm_valtype_new(wasm_valkind_t kind)
 {
     wasm_valtype_t *val_type;
 
+    if (kind > WASM_F64 && WASM_ANYREF != kind && WASM_FUNCREF != kind) {
```

Comment:
Shall we add `#if WASM_ENABLE_REF_TYPES != 0` macro control for `WASM_ANYREF`? Like the change below, L792?